File: src/main/java/top/zbeboy/isy/service/UsersTypeServiceImpl.java
package top.zbeboy.isy.service;

import org.jooq.Configuration;
import org.jooq.DSLContext;
import org.jooq.Result;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Propagation;
import org.springframework.transaction.annotation.Transactional;
import top.zbeboy.isy.domain.tables.daos.UsersTypeDao;
import top.zbeboy.isy.domain.tables.pojos.Users;
import top.zbeboy.isy.domain.tables.pojos.UsersType;
import top.zbeboy.isy.domain.tables.records.UsersTypeRecord;

import javax.annotation.Resource;

import static top.zbeboy.isy.domain.tables.UsersType.USERS_TYPE;

/**
 * Created by lenovo on 2016-08-24.
 */
@Service("usersTypeService")
@Transactional(propagation = Propagation.SUPPORTS, readOnly = true)
public class UsersTypeServiceImpl implements UsersTypeService {

    private final Logger log = LoggerFactory.getLogger(UsersTypeServiceImpl.class);

    private final DSLContext create;

    @Resource
    private UsersTypeDao usersTypeDao;

    @Resource
    private UsersService usersService;

    @Autowired
    public UsersTypeServiceImpl(DSLContext dslContext) {
        this.create = dslContext;
    }


    @Override
    public UsersType findByUsersTypeName(String usersTypeName) {
        return usersTypeDao.fetchOne(USERS_TYPE.USERS_TYPE_NAME, usersTypeName);
    }

    @Override
    public UsersType findByUsersTypeId(int usersTypeId) {
        return usersTypeDao.findById(usersTypeId);
    }

    @Override
    public Result<UsersTypeRecord> findAll() {
        return create.selectFrom(USERS_TYPE).fetch();
    }

    @Override
    public boolean isCurrentUsersTypeName(String usersTypeName) {
        Users users = usersService.getUserFromSession();
        String usersType = usersTypeDao.fetchOneByUsersTypeId(users.getUsersTypeId()).getUsersTypeName();
        return usersTypeName.equals(usersType);
    }
}

File: src/main/java/top/zbeboy/isy/service/platform/UsersTypeServiceImpl.kt
package top.zbeboy.isy.service.platform

import org.jooq.DSLContext
import org.jooq.Result
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Propagation
import org.springframework.transaction.annotation.Transactional
import top.zbeboy.isy.domain.tables.daos.UsersTypeDao
import top.zbeboy.isy.domain.tables.records.UsersTypeRecord
import javax.annotation.Resource

import top.zbeboy.isy.domain.tables.UsersType.USERS_TYPE
/**
 * Created by zbeboy 2017-11-16 .
 **/
@Service("usersTypeService")
@Transactional(propagation = Propagation.SUPPORTS, readOnly = true)
open class UsersTypeServiceImpl @Autowired constructor(dslContext: DSLContext) : UsersTypeService {

    private val create: DSLContext = dslContext

    @Resource
    open lateinit var usersTypeDao: UsersTypeDao

    @Resource
    open lateinit var usersService: UsersService

    override fun findAll(): Result<UsersTypeRecord> {
        return create.selectFrom<UsersTypeRecord>(USERS_TYPE).fetch()
    }

    override fun isCurrentUsersTypeName(usersTypeName: String): Boolean {
        val users = usersService.getUserFromSession()
        val usersType = usersTypeDao.fetchOneByUsersTypeId(users!!.usersTypeId).usersTypeName
        return usersTypeName == usersType
    }
}